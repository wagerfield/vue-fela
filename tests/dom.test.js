import { createRenderer } from 'fela'
import {
  mapStyle,
  renderServerStyles,
  renderClientStyles
} from '../lib/dom'

const fela = createRenderer()

const renderStyle = (type, css, media, support) =>
  mapStyle({ type, css, media, support })

const rule1 = ({ width }) => ({ width })
const rule2 = ({ color }) => ({ color })
const rule3 = ({ width, color = 'red' }) => ({
  [`@media (min-width: ${width})`]: { color }
})

describe('DOM', () => {

  beforeEach(fela.clear)

  describe('mapStyle(style)', () => {

    it('renders expected snapshots', () => {
      const css = '.a{color:black}'
      const media = '(min-width:300px)'
      expect(renderStyle()).toMatchSnapshot()
      expect(renderStyle('RULE', css)).toMatchSnapshot()
      expect(renderStyle('STATIC', css)).toMatchSnapshot()
      expect(renderStyle('RULE', css, media)).toMatchSnapshot()
      expect(renderStyle('RULE', css, media, true)).toMatchSnapshot()
      expect(renderStyle('RULE', css, false, true)).toMatchSnapshot()
    })
  })

  describe('renderServerStyles(renderer)', () => {

    it('renders expected snapshot', () => {

      fela.renderRule(rule1, { width: '8px' })
      fela.renderRule(rule2, { color: 'red' })
      fela.renderRule(rule3, { width: '8px' })

      fela.renderFont('Lato', [
        '../fonts/Lato.ttf',
        '../fonts/Lato.eot'
      ], { fontWeight: 300 })

      fela.renderStatic('body{margin:0}')
      fela.renderStatic({
        boxSizing: 'border-box'
      }, '*')

      const styles = renderServerStyles(fela)
      expect(styles).toMatchSnapshot()
    })
  })

  describe('renderClientStyles(renderer)', () => {

    it('calls fela dom rehydrate and render methods', () => {
      expect(fela.updateSubscription).toBeUndefined()
      expect(fela.listeners).toHaveLength(0)

      renderClientStyles(fela)

      expect(fela.updateSubscription).toBeDefined()
      expect(fela.listeners).toHaveLength(1)
    })
  })
})
