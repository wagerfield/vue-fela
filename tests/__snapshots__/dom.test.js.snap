// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM mapStyle(style) renders expected snapshots 1`] = `
Object {
  "once": true,
  "type": "text/css",
}
`;

exports[`DOM mapStyle(style) renders expected snapshots 2`] = `
Object {
  "cssText": ".a{color:black}",
  "data-fela-type": "RULE",
  "hid": "rule",
  "once": true,
  "type": "text/css",
}
`;

exports[`DOM mapStyle(style) renders expected snapshots 3`] = `
Object {
  "cssText": ".a{color:black}",
  "data-fela-type": "STATIC",
  "hid": "static",
  "once": true,
  "type": "text/css",
}
`;

exports[`DOM mapStyle(style) renders expected snapshots 4`] = `
Object {
  "cssText": ".a{color:black}",
  "data-fela-type": "RULE",
  "hid": "rule(min-width:300px)",
  "media": "(min-width:300px)",
  "once": true,
  "type": "text/css",
}
`;

exports[`DOM mapStyle(style) renders expected snapshots 5`] = `
Object {
  "cssText": ".a{color:black}",
  "data-fela-support": true,
  "data-fela-type": "RULE",
  "hid": "rule[support](min-width:300px)",
  "media": "(min-width:300px)",
  "once": true,
  "type": "text/css",
}
`;

exports[`DOM mapStyle(style) renders expected snapshots 6`] = `
Object {
  "cssText": ".a{color:black}",
  "data-fela-support": true,
  "data-fela-type": "RULE",
  "hid": "rule[support]",
  "once": true,
  "type": "text/css",
}
`;

exports[`DOM renderServerStyles(renderer) renders expected snapshot 1`] = `
Array [
  Object {
    "cssText": "@font-face{font-weight:300;src:url('../fonts/Lato.ttf') format('truetype'),url('../fonts/Lato.eot') format('embedded-opentype');font-family:\\"Lato\\"}",
    "data-fela-rehydration": 3,
    "data-fela-type": "FONT",
    "hid": "font",
    "once": true,
    "type": "text/css",
  },
  Object {
    "cssText": "body{margin:0}*{box-sizing:border-box}",
    "data-fela-rehydration": 3,
    "data-fela-type": "STATIC",
    "hid": "static",
    "once": true,
    "type": "text/css",
  },
  Object {
    "cssText": ".a{width:8px}.b{color:red}",
    "data-fela-rehydration": 3,
    "data-fela-type": "RULE",
    "hid": "rule",
    "once": true,
    "type": "text/css",
  },
  Object {
    "cssText": ".c{color:red}",
    "data-fela-rehydration": 3,
    "data-fela-type": "RULE",
    "hid": "rule(min-width: 8px)",
    "media": "(min-width: 8px)",
    "once": true,
    "type": "text/css",
  },
]
`;
